Xamarin.Forms.Platform.WinRT/FormsTextBox.cs

```diff
@@ -237,19 +240,75 @@ static void OnIsPasswordChanged(DependencyObject dependencyObject, DependencyPro
 
 		void OnSelectionChanged(object sender, RoutedEventArgs routedEventArgs)
 		{
-			if (!IsPassword)
-			{
-				return;
-			}
+			// Cache this value for later use as explained in OnKeyDown below
+			_cachedSelectionLength = SelectionLength;
+		}
 
-			// Prevent the user from selecting any text in the password box by forcing all selection
-			// to zero-length at the end of the Text
-			// This simulates the "do not allow clipboard copy" behavior the PasswordBox control has
-			if (SelectionLength > 0 || SelectionStart < Text.Length)
+		// Because the implementation of a password entry is based around inheriting from TextBox (via FormsTextBox), there
+		// are some inaccuracies in the behavior. OnKeyDown is what needs to be used for a workaround in this case because 
+		// there's no easy way to disable specific keyboard shortcuts in a TextBox, so key presses are being intercepted and 
+		// handled accordingly.
+		protected override void OnKeyDown(KeyRoutedEventArgs e)
+		{
+			if (IsPassword)
 			{
-				SelectionLength = 0;
-				SelectionStart = Text.Length;
+				// The _ctrlKeyActive flag is used to track if that key is pressed; we want to pretend Ctrl is not actively being
+				// used by treating the key routing as handled.
+				if (e.Key == VirtualKey.Control)
+				{
+					if (e.Key == VirtualKey.Control)
+					{
+						_ctrlKeyActive = true;
```

Comment:
Rather than tracking this with a member variable, you could probably just do something like
`var ctrl = Window.Current.CoreWindow.GetKeyState(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);` here; then you don't have to worry about `OnKeyUp`.